reflect.go

```diff
@@ -120,8 +121,54 @@ func runInDir(program []byte, dir string) (*model.PackedPkg, error) {
 		return nil, err
 	}
 
+	{
+		// Copy go.mod into the build directory:
+		wd, err := os.Getwd()
+		if err != nil {
+			log.Fatalf("Unable to load current directory: %v", err)
+		}
+
+		modRoot := findModuleRoot(wd)
+
+		if modRoot != "" {
+			MustCopyFile(filepath.Join(modRoot, "go.mod"), filepath.Join(tmpDir, "go.mod"))
+		}
+
+	}
+
 	return run(filepath.Join(tmpDir, progBinary))
 }
+func MustCopyFile(src, dst string) {
+	_, err := copyFile(src, dst)
+	if err != nil {
+		log.Fatalf("error copying %q to %q: %s", src, dst, err)
+	}
+}
+
+func copyFile(src, dst string) (int64, error) {
```

Comment:
Why not just use `io.Copy`?

Reply:
`copyFile` does use `io.Copy` under the hood.

`io.Copy` does not allow to directly do the whole file creation and copy: here's the signature of `io.Copy`: 

`func Copy(dst Writer, src Reader) (written int64, err error)`

Reply:
:facepalm: yes, I failed to read.